src/vector.h

```diff
@@ -0,0 +1,35 @@
+#ifndef _VECTOR_H
+#define _VECTOR_H
+
+#include <Python.h>
+#include <numpy/ndarraytypes.h>
+
+#include "geos.h"
```

Comment:
Only put includes in a header if the header itself needs them. I think you should drop ` geos.h` and `kvec.h` here, only `pygeom.h` is needed for the `GeometryObject` .